src/net.c

```diff
@@ -821,10 +821,10 @@ coap_send_pdu(coap_session_t *session, coap_pdu_t *pdu, coap_queue_t *node) {
     (session->sock.flags & COAP_SOCKET_WANT_WRITE))
     return coap_session_delay_pdu(session, pdu, node);
 
-  if (pdu->type == COAP_MESSAGE_CON && COAP_PROTO_NOT_RELIABLE(session->proto))
-    session->con_active++;
-
   bytes_written = coap_session_send_pdu(session, pdu);
+  if (bytes_written >= 0 && pdu->type == COAP_MESSAGE_CON &&
```

Comment:
I think the same change then needs to go into the `WITH_LWIP` part as well (see line 738).

Reply:
Good catch - code updated and pushed.